optuna/study/study.py

```diff
@@ -1411,4 +1411,47 @@ def objective(trial):
     """
 
     storage = storages.get_storage(storage)
-    return storage.get_all_study_summaries(include_best_trial=include_best_trial)
+    frozen_studies = storage.get_all_studies()
+    study_summaries = []
+
+    for s in frozen_studies:
+
+        all_trials = storage.get_all_trials(s._study_id)
+        completed_trials = [t for t in all_trials if t.state == TrialState.COMPLETE]
+
+        n_trials = len(all_trials)
+
+        if len(s.directions) == 1:
+            direction = s.direction
+            directions = None
+            if include_best_trial and len(completed_trials) != 0:
+                if direction == StudyDirection.MAXIMIZE:
+                    best_trial = max(completed_trials, key=lambda t: cast(float, t.value))
+                else:
+                    best_trial = min(completed_trials, key=lambda t: cast(float, t.value))
```

Comment:
Can we use `storage.get_best_trial()` here?

Reply:
`storage.get_all_trials()` is called in line L.1414, which loads all information of trials into memory. That said, current implementation is more performant than calling `storage.get_best_trial()` because it does not have to reach the storage again.

Reply:
Sounds reasonable 👍  Thanks for the clarification.